src/extension.ts

```diff
@@ -37,17 +37,43 @@ const crcLogProvider = new LogProvider(commander);
 const defaultStatus = { CrcStatus: 'Unknown', Preset: 'Unknown' };
 
 export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
-  // crc is installed or not ?
-  if (!fs.existsSync(crcBinary())) {
-    console.warn('Can not find CRC binary!');
-    return;
+  const crcVersion = await getCrcVersion();
+
+  const detectionChecks: extensionApi.ProviderDetectionCheck[] = [];
+  let status: extensionApi.ProviderStatus = 'not-installed';
+  let preset = 'unknown';
+  if (crcVersion) {
+    status = 'installed';
+
+    const daemonStarted = await daemonStart();
+    if (!daemonStarted) {
+      //TODO handle this
```

Comment:
we have `daemonStart()`, right?

Reply:
yes, there https://github.com/crc-org/crc-extension/pull/4/files#diff-1d4ce93c62fdee003099debe7bef7cbcbc47d037c8e5b86f300f3a11b39dc23eR136